### ISSUE TITLE
Endless loading after adding and then switching to newly added network

### ISSUE THREAD

**Issue by DenysRakhuba** [MetaMask/metamask-mobile#7259]
**Describe the bug**
I've been currently working on adding support of mobile Metamask in our app and have faced a problem with adding a new network (Polygon Matic). When user tries to connect to a network, that has not been added yet to Metamask, he receives a relevant prompt dialog. After he confirms adding it, he's getting another prompt dialog asking if he would like to switch to the newly added network and when he clicks "Yes", endless loading appears, while on the background it's seen that application was able to get user account from Metamask. The only way to handle this issue is to close the wallet and open it again. Then user sees that new network were added and he can connect without any issues. 

**Screenshots**
<img width="375" alt="Screen Shot 2023-09-18 at 1 14 27 PM" src="https://github.com/MetaMask/metamask-mobile/assets/16452697/f104c936-bc8a-44a9-91f3-f7e4eff4a987">
<img width="381" alt="Screen Shot 2023-09-18 at 1 16 41 PM" src="https://github.com/MetaMask/metamask-mobile/assets/16452697/6fe8c3e0-39b6-40c9-a0bd-79fe0cdcf9f4">
<img width="379" alt="Screen Shot 2023-09-18 at 1 16 53 PM" src="https://github.com/MetaMask/metamask-mobile/assets/16452697/3d6c82d0-a921-4d9c-8825-1a967667c2bb">

**To Reproduce**
Just try to connect to network that's not in Metamask yet. In our case it's Polygon Matic. 
Here is part of the code responsible for this logic, if it helps somehow:

   ```
 if (numberToHex(networkVersion) !== this.chainParams.chainId && !ignoreCheckNetworks) {
      if (autoConnection) {
        return;
      }
      try {
        try {
          await this.ethereum.request({
            method: 'wallet_switchEthereumChain',
            params: [{chainId: this.chainParams.chainId}]
          });
        } catch (e: any) {
          const code = e.code;
          if (code && e.code === 4902) {
            await this.ethereum.request({
              method: 'wallet_addEthereumChain',
              params: [this.chainParams]
            });
          } else {
            throw this.error(CONNECTION_MESSAGES.SWITCH_NETWORK);
          }
        }
      } catch (e) {
        throw this.error(CONNECTION_MESSAGES.SWITCH_NETWORK);
      }
    }
    const accounts: string[] = await this.ethereum.request({method: 'eth_requestAccounts'});
```

**Expected behavior**
User should be able to continue using application without endless loader. 

**Smartphone (please complete the following information):**
 - Device: Google Pixel 7 Pro
 - OS: Android 13
 - App Version: v7.6.0 (1169)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by cryptotavares** [comment]
@DenysRakhuba The team was looking into it and they were only able to reproduce the described behaviour when two add network approval requests exist at the same time. That was fixed on this [PR](https://github.com/MetaMask/metamask-mobile/pull/7276) and should be available on the upcoming 7.9.0 version of the app.

Can you confirm that you are not triggering the add network request twice?

**Comment by DenysRakhuba** [comment]
@cryptotavares there's only one add network request from our side. But right after it there's switch network request. I'll check with a new version of the application if this problem still exists and let you know if so. Thank you very much for a quick response and handling it!
